pyproject.toml

```diff
@@ -11,6 +11,8 @@ requires = [
     # see: https://github.com/scipy/oldest-supported-numpy/blob/main/setup.cfg
     "oldest-supported-numpy",
 
+    # nightly scipy 1.10.0.dev0 are the first wheels available for python 3.11
+    "scipy==1.10.0.dev0; python_version>='3.11'",
```

Comment:
Naive question: If we rely on nightly builds of SciPy which uses [this alternative index](https://pypi.anaconda.org/scipy-wheels-nightly/simple/), must we publish wheels for Python 3.11 on this index uniquely? (I am not familiar with the CD enough to know if it's the case.)

Reply:
I think our Python 3.11 wheel should only be on the alternative index until SciPy and NumPy releases a Python 3.11 wheel on PyPI. (If NumPy and SciPy does not have 3.11 wheels on PyPI, then users with 3.11 would need to build NumPy and SciPy when installing the scikit-learn wheel.)

Once SciPy and NumPy release Python 3.11 wheels on PyPI, we can upload wheels there too.

**Implementation-wise:** When this PR is merged to build 3.11 wheels, there are two releasing situations. It depends on if scikit-learn 1.2 is released before or after SciPy and NumPy wheels are on PyPI for Python 3.11:

- **before**: Update `build_tools/github/check_wheels.py` to ignore a specific version of Python when counting the number of wheels, likely through a cli argument. **This can be done in a follow up PR** (We may need to do this anyway so we can build and test with Python-dev but also make a release before the Python-dev version is released)
- **after**: Nothing needs to change.